false-negative/infinite-recursion.cpp

```diff
@@ -0,0 +1,12 @@
+// Clang-Tidy cannot detect infinite recursion.
+
+#include <iostream>
+
+// This function contains infinite recursion.
+int recursion(int x) {
+    return recursion(x + 1) + recursion(x - 1);
```

Comment:
@andrewt0301 if you remove the second operand, will the problem remain?

```
int recursion(int x) {
  return recursion(x+1);
}
```

Reply:
I have updated the example. The problem does remain. This means that even trivial cases are not covered by Clang-Tidy.